Menus/SettingsVideoGUI.h

```diff
@@ -0,0 +1,196 @@
+
+			/// <summary>
+			/// Comparison operator for eliminating duplicates and sorting in the temporary PresetResolutionRecord std::sets during PopulateResolutionsComboBox.
+			/// </summary>
+			/// <param name="rhs">The PresetResolutionRecord to compare with.</param>
+			/// <returns>Bool with the result of the comparison.</returns>
+			bool operator<(const PresetResolutionRecord &rhs) const {
+				if (Width == rhs.Width && Height == rhs.Height) {
+					return Upscaled != rhs.Upscaled;
+				} else if (Width == rhs.Width) {
+					return Height < rhs.Height;
+				}
+				return Width < rhs.Width;
+			}
+		};
+
+		GUIControlManager *m_GUIControlManager; //!< The GUIControlManager which holds all the GUIControls of this menu. Not owned by this.
+
+		std::vector<PresetResolutionRecord> m_PresetResolutions; //!< Contains PresetResolutionRecords for all the supported preset resolutions.
+
+		int m_NewGraphicsDriver; //!< The new graphics driver to use when changing resolution.
+		int m_NewResX; //!< The new resolution width to use when changing resolution.
+		int m_NewResY; //!< The new resolution height to use when changing resolution.
+		bool m_NewResUpscaled; //!< Whether the new resolution should be upscaled when changing resolution.
+
+		/// <summary>
+		/// GUI elements that compose the video settings menu screen.
+		/// </summary>
+		GUICollectionBox *m_VideoSettingsBox;
+		GUIRadioButton *m_TwoPlayerSplitscreenHSplitRadioButton;
+		GUIRadioButton *m_TwoPlayerSplitscreenVSplitRadioButton;
+		GUIRadioButton *m_PresetResolutionRadioButton;
+		GUIRadioButton *m_CustomResolutionRadioButton;
+		GUICollectionBox *m_PresetResolutionBox;
+		GUIComboBox *m_PresetResolutionComboBox;
+		GUIButton *m_PresetResolutionApplyButton;
+		GUILabel *m_PresetResolutionMessageLabel;
+		GUICollectionBox *m_CustomResolutionBox;
+		GUITextBox *m_CustomResolutionWidthTextBox;
+		GUITextBox *m_CustomResolutionHeightTextBox;
+		GUICheckbox *m_CustomResolutionUpscaledCheckbox;
+		GUIRadioButton *m_CustomResolutionBorderlessRadioButton;
+		GUIRadioButton *m_CustomResolutionDedicatedRadioButton;
+		GUILabel *m_CustomResolutionMessageLabel;
+		GUIButton *m_CustomResolutionApplyButton;
+		GUICollectionBox *m_ResolutionChangeDialogBox;
+		GUIButton *m_ResolutionChangeConfirmButton;
+		GUIButton *m_ResolutionChangeCancelButton;
+		std::array<GUIButton *, ResolutionQuickChangeType::QuickChangeTypeCount> m_ResolutionQuickToggleButtons;
+
+#pragma region Create Breakdown
+		/// <summary>
+		/// Creates all the elements that compose the preset resolution selection box.
+		/// </summary>
+		void CreatePresetResolutionBox();
+
+		/// <summary>
+		/// Creates all the elements that compose the custom resolution controls box.
+		/// </summary>
+		void CreateCustomResolutionBox();
+#pragma endregion
+
+#pragma region Video Settings Handling
+		/// <summary>
+		/// Checks whether the passed in width and height values make a supported resolution setting.
```

Comment:
It doesn't try to match with anything.
I guess this needs to be reworded.

Reply:
Yeah, could do with rewording.

Reply:
Yeah I don't know how to reword it. ¯\\\_( ͡° ͜ʖ ͡°)_/¯

Reply:
Since, as you say, it's not matching it against a list or something, it's instead checking it against a bunch of rules (if-elses), maybe something like:
`Gets whether or not the passed in width and height values satisfy our supported resolution rules.`

I'm not entirely sure here either lol